File: Tests/SignalR.Extras.Autofac.Test/stubs/GenericLifetimeHubStub.fs
namespace SignalR.Extras.Autofac.Test.Stubs

open SignalR.Extras.Autofac
open SignalR.Extras.Autofac.Test.Stubs.Dependencies

type GenericLifetimeHubStub(dep1 : ScopedObjectStub, dep2 : SingletonObjectStub) =
    inherit LifetimeHub<IHubClientStub>()
    member this.ScopedDependency = dep1
    member this.SingletonDependency = dep2

File: Tests/SignalR.Extras.Autofac.Tests/Stubs/GenericLifetimeHubStub.cs
using SignalR.Extras.Autofac.Tests.Stubs.Dependencies;

namespace SignalR.Extras.Autofac.Tests.Stubs;

public class GenericLifetimeHubStub : LifetimeHub<IHubClientStub>
{

    public GenericLifetimeHubStub(ScopedObjectStub dep1, SingletonObjectStub dep2, RequestScopedObjectStub dep3)
    {
        ScopedDependency = dep1;
        SingletonDependency = dep2;
        RequestScopedDependency = dep3;
    }

    public readonly ScopedObjectStub ScopedDependency;
    public readonly SingletonObjectStub SingletonDependency;
    public readonly RequestScopedObjectStub RequestScopedDependency;

}